### PR TITLE
orx-kinect refactoring + new general orx-depth-camera

### PR DESCRIPTION
**Pull request by morisil**
I changed a lot with this PR and unfortunately I couldn't keep down compatibility:
* new `orx-depth-camera` interface for generalized API of all the depth cameras
* 3 modes of depth measurement: RAW_NORMALIZED, METERS, RAW
* initial low level kinect shaders operating on integer values, not floating point values 
* the concept of `depthCamera.frameFlow: Flow<ColorBuffer>` operating on drawer thread coroutines for async processing of kinect frames
* extensive set of demo programs to show possible use cases
* colormap filters externalized as mpp filters in `orx-fx`
* `depthCamera.mirror` property renamed to `flipH` and corresponding `flipV` added (I have use cases with projections calibrated over the floor level where it is very useful).

### CODE REVIEW

File: orx-depth-camera/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+plugins {
+    kotlin("multiplatform")
+}
+
+kotlin {
+    jvm {
+        compilations.all {
+            kotlinOptions.jvmTarget = libs.versions.jvmTarget.get()
+            kotlinOptions.apiVersion = libs.versions.kotlinApi.get()
+        }
```
Comment (Vechro) [inline orx-depth-camera/build.gradle.kts:10]: Conflicts with #256. If #256 is merged first, the `compilations.all` block should be removed in this PR.

File: orx-jvm/orx-kinect-common/build.gradle [hunk 2]
```diff
@@ -0,0 +1,3 @@
+dependencies {
+    api project(":orx-depth-camera")
```
Comment (Vechro) [inline orx-jvm/orx-kinect-common/build.gradle:2]: I'd recommend using Kotlin DSL if you're going to be writing a new build script

Comment (morisil) [reply orx-jvm/orx-kinect-common/build.gradle:2]: Good point, although surrounding scripts are still `.gradle` I didn't want to accidentally brake something. I will check if migrating to `.gradle.kts` now brings any side effects.

File: orx-depth-camera/build.gradle.kts (hunk 1 above)
Comment (morisil) [reply orx-depth-camera/build.gradle.kts:10]: @edwinRNDR mentioned that 0.4.1 should go first. So probably I will update this PR once it is released.

File: orx-jvm/orx-kinect-common/build.gradle (hunk 2 above)
Comment (Vechro) [reply orx-jvm/orx-kinect-common/build.gradle:2]: I'm pretty sure you can rewrite the second line as `api(project(":orx-depth-camera"))`, then rename the file to `build.gradle.kts` and you'll have an equivalent build script.

Comment (morisil) [reply orx-jvm/orx-kinect-common/build.gradle]: done

File: orx-depth-camera/build.gradle.kts (hunk 1 above)
Comment (morisil) [reply orx-depth-camera/build.gradle.kts]: @Vechro done

**Comment by hamoid** [comment]
Hi :) Is it expected that `orx-depth-camera` and `orx-depth-camera-calibrator` could be used independently? Or always together? Just wondering if the second could be part of the first (since it seems to be only one file) to avoid creating more orx'es than needed.

Great job, looks like very useful for all of you using depth cameras :)

**Comment by morisil** [comment]
@hamoid My intent was to make `orx-depth-camera` as an API independent from the implementation, therefore it's MPP module, potentially capable of supporting something like webcam from the browser with some ML depth estimation. The calibrator is dependent on `orx-parameters` and the logic of storing parameters between program runs, which works only on jvm, even though `orx-parameters` is technically MPP, but making it work in JS would required, I believe, providing some form of annotation processor, as reflection is not suppoorted. So I can see use cases of these 2 being used independently. But technically it would be possible to merge them, as `orx-parameters` is a MPP module, even if it's not fully implemented.

**Comment by morisil** [comment]
@hamoid , ok, just checked, not possible at the moment because `orx-depth-camera-calibrator` depends on `:orx-jvm:orx-gui` :( Once we have JS version of GUI, which I would be actually happy to have, then it can be merged.

**Comment by hamoid** [comment]
Thank you for trying! :-)

**Pull request merged**